torchtune/training/checkpointing/_checkpointer.py

```diff
@@ -374,27 +382,125 @@ def __init__(
             Path.joinpath(self._checkpoint_dir, "config.json").read_text()
         )
 
-        # save config.json to output_dir
-        save_config(self._output_dir, self._config)
+        # save all files in input_dir, except model weights and mapping, to output_dir
+        # this is useful to preserve the tokenizer, configs, license, etc.
+        copy_files(
+            self._checkpoint_dir,
+            Path.joinpath(self._output_dir, BASE_MODEL_DIRNAME),
+            ignore_suffixes=training.SUFFIXES_TO_NOT_COPY,
+        )
 
-        # recipe_checkpoint contains the recipe state. This should be available if
-        # resume_from_checkpoint is True
-        self._recipe_checkpoint = None
-        if self._resume_from_checkpoint:
-            if recipe_checkpoint is None:
-                raise ValueError(
-                    "If resume_from_checkpoint is True, recipe_checkpoint file must be provided."
+        # repo_id is necessary for when saving an adapter config, so its compatible with HF.
+        # This json file is produced and saved in the download step.
+        # contents are {"repo_id": "some_model/some_model_version"}
+        repo_id_path = Path.joinpath(
```

Comment:
Again I think this can just be joined with a forward-slash ?

Reply:
Easier to read IMO and I think we use it in other places, but es okay.

Reply:
keeping it for safety